File: server/src/main/kotlin/pt/isel/g20/unicommunity/blackboardItem/model/BlackboardItem.java
package pt.isel.g20.unicommunity.blackboardItem.model;

import javax.persistence.*;

@Entity
public class BlackboardItem {

    @Column
    private long boardId;

    @Column
    private long bbId;

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private long id;

    @Column
    private String name;

    @Column
    private String content;

    public BlackboardItem(){}

    public BlackboardItem(long boardId, long bbId, String name, String content){
        this.boardId = boardId;
        this.bbId = bbId;
        this.name = name;
        this.content = content;
    }

    public BlackboardItem(long boardId, long bbId, long id, String name, String content){
        this.boardId = boardId;
        this.bbId = bbId;
        this.id = id;
        this.name = name;
        this.content = content;
    }

    public long getBoardId() {
        return boardId;
    }

    public void setBoardId(int boardId) {
        this.boardId = boardId;
    }

    public long getBbId() {
        return bbId;
    }

    public void setBbId(int bbId) {
        this.bbId = bbId;
    }

    public long getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }
}

File: server/src/main/kotlin/pt/isel/g20/unicommunity/blackboardItem/model/BlackboardItem.kt
package pt.isel.g20.unicommunity.blackboardItem.model

import com.fasterxml.jackson.annotation.JsonFormat
import com.fasterxml.jackson.annotation.JsonIgnore
import org.hibernate.annotations.CreationTimestamp
import pt.isel.g20.unicommunity.blackboard.model.Blackboard
import pt.isel.g20.unicommunity.user.model.User
import java.util.*
import javax.persistence.*

@Entity
class BlackboardItem(
        @Column var name: String,
        @Column var content: String,
        @ManyToOne @JoinColumn var author: User,
        @JsonIgnore @ManyToOne var blackboard: Blackboard
) {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long = 0

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd@HH:mm:ss.SSSZ", locale = "en_GB")
    @Column
    @CreationTimestamp
    var createdAt: Date? = null
}
